lib/Widgets/TerminalView.vala

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2025 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+[Version (since = "7.7.0")]
+public class Granite.TerminalView : Granite.Bin {
+
+    public bool autoscroll { get; set; default = false; }
+    public string text { get; set; }
+
+    private Gtk.TextBuffer buffer;
+    private double prev_upper_adj = 0;
+    private Gtk.ScrolledWindow scrolled_window;
+
+    construct {
+        var view = new Gtk.TextView () {
+            cursor_visible = false,
+            editable = false,
+            monospace = true,
+            pixels_below_lines = 3,
+            left_margin = 9,
+            right_margin = 9,
+            top_margin = 6,
+            bottom_margin = 6,
+            wrap_mode = Gtk.WrapMode.WORD
+        };
+
+        buffer = view.get_buffer ();
+
+        scrolled_window = new Gtk.ScrolledWindow () {
+            child = view,
+            hexpand = true,
+            vexpand = true,
+            hscrollbar_policy = NEVER,
+        };
+
+        child = scrolled_window;
+        add_css_class (Granite.CssClass.TERMINAL);
```

Comment:
Instead of adding a style class, this should be the css name. Right now you have css nodes like `widget.terminal` but it should just be `terminal` So you need to remove this css class and add something like:
```
    class construct {
        set_css_name ("terminal-view");
    }
```